commcare_connect/reports/views.py

```diff
@@ -359,15 +359,32 @@ def __init__(self, *args, **kwargs):
             ),
             Row(
                 Column("delivery_type", css_class="col-md-4"),
-                Column("year", css_class="col-md-4"),
-                Column("month", css_class="col-md-4"),
+                Column("from_date", css_class="col-md-4"),
+                Column("to_date", css_class="col-md-4"),
             ),
-            Row(Column("by_delivery_type", css_class="col-md-4")),
+            Row(Column("group_by_delivery_type", css_class="col-md-4")),
         )
 
+        if not self.data:
+            self.data = self.data.copy() if self.data else {}
+            today = date.today()
+            default_from = today - timedelta(days=30)
+            self.data["to_date"] = today.strftime("%Y-%m")
+            self.data["from_date"] = default_from.strftime("%Y-%m")
+            self.form.is_bound = True
```

Comment:
where is `self.form` defined and can we use it to do the validation that the `to_date` must be later rather than just ignoring it in the data fetch method if it is invalid

Reply:
This form is configured automatically from the fields provided on the FilterSet.